crates/signer-ledger/src/signer.rs

```diff
@@ -43,7 +46,33 @@ impl alloy_network::TxSigner<Signature> for LedgerSigner {
         &self,
         tx: &mut dyn SignableTransaction<Signature>,
     ) -> Result<Signature> {
-        sign_transaction_with_chain_id!(self, tx, self.sign_tx_rlp(&tx.encoded_for_signing()).await)
+        let encoded = tx.encoded_for_signing();
+
+        match encoded.as_slice() {
+            // Ledger requires passing EIP712 data to a separate instruction
+            #[cfg(feature = "eip712")]
+            [0x19, 0x1, data @ ..] => {
```

Comment:
I don't this would clash with the other txs, so this seems okay to me.

Reply:
I think [`ERC-191`](https://github.com/ethereum/EIPs/issues/191) describes the format and mentions that `0x19` is an invalid RLP header so no valid Ethereum txs will ever start with `0x19`, and then the `0x1` is the version specifier for `EIP712`, so yeah shouldn't ever conflict